### ISSUE TITLE
Doesn't generate screenshot for some pages if it has broken css styles

### ISSUE THREAD

**Issue by VitalyOrbit** [ariya/phantomjs#14160]
First of all, I test with local resources in vagrant.
The style I found in css was like this: **background: url() no-repeat;**
Contents cannot be checked.

Here is some debug info:
```
[vagrant@vagrant u1]$ /vagrant/phantomjs --debug="true" --ignore-ssl-errors="true" '/vagrant/templates2.js'
2016-04-01T12:10:47 [DEBUG] CookieJar - Created but will not store cookies (use option '--cookies-file=<filename>' to enable persistent cookie storage)
2016-04-01T12:10:47 [DEBUG] Set  "http"  proxy to:  "" : 1080
2016-04-01T12:10:47 [DEBUG] Phantom - execute: Configuration
2016-04-01T12:10:47 [DEBUG]      0 objectName : ""
2016-04-01T12:10:47 [DEBUG]      1 cookiesFile : ""
2016-04-01T12:10:47 [DEBUG]      2 diskCacheEnabled : "false"
2016-04-01T12:10:47 [DEBUG]      3 maxDiskCacheSize : "-1"
2016-04-01T12:10:47 [DEBUG]      4 diskCachePath : ""
2016-04-01T12:10:47 [DEBUG]      5 ignoreSslErrors : "true"
2016-04-01T12:10:47 [DEBUG]      6 localUrlAccessEnabled : "true"
2016-04-01T12:10:47 [DEBUG]      7 localToRemoteUrlAccessEnabled : "false"
2016-04-01T12:10:47 [DEBUG]      8 outputEncoding : "UTF-8"
2016-04-01T12:10:47 [DEBUG]      9 proxyType : "http"
2016-04-01T12:10:47 [DEBUG]      10 proxy : ":1080"
2016-04-01T12:10:47 [DEBUG]      11 proxyAuth : ":"
2016-04-01T12:10:47 [DEBUG]      12 scriptEncoding : "UTF-8"
2016-04-01T12:10:47 [DEBUG]      13 webSecurityEnabled : "true"
2016-04-01T12:10:47 [DEBUG]      14 offlineStoragePath : ""
2016-04-01T12:10:47 [DEBUG]      15 localStoragePath : ""
2016-04-01T12:10:47 [DEBUG]      16 localStorageDefaultQuota : "-1"
2016-04-01T12:10:47 [DEBUG]      17 offlineStorageDefaultQuota : "-1"
2016-04-01T12:10:47 [DEBUG]      18 printDebugMessages : "true"
2016-04-01T12:10:47 [DEBUG]      19 javascriptCanOpenWindows : "true"
2016-04-01T12:10:47 [DEBUG]      20 javascriptCanCloseWindows : "true"
2016-04-01T12:10:47 [DEBUG]      21 sslProtocol : "default"
2016-04-01T12:10:47 [DEBUG]      22 sslCiphers : "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-RC4-SHA:ECDHE-RSA-RC4-SHA:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:RC4-SHA:RC4-MD5"
2016-04-01T12:10:47 [DEBUG]      23 sslCertificatesPath : ""
2016-04-01T12:10:47 [DEBUG]      24 sslClientCertificateFile : ""
2016-04-01T12:10:47 [DEBUG]      25 sslClientKeyFile : ""
2016-04-01T12:10:47 [DEBUG]      26 sslClientKeyPassphrase : ""
2016-04-01T12:10:47 [DEBUG]      27 webdriver : ":"
2016-04-01T12:10:47 [DEBUG]      28 webdriverLogFile : ""
2016-04-01T12:10:47 [DEBUG]      29 webdriverLogLevel : "INFO"
2016-04-01T12:10:47 [DEBUG]      30 webdriverSeleniumGridHub : ""
2016-04-01T12:10:47 [DEBUG] Phantom - execute: Script & Arguments
2016-04-01T12:10:47 [DEBUG]      script: "/vagrant/templates2.js"
2016-04-01T12:10:47 [DEBUG] Phantom - execute: Starting normal mode
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/fs.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/system.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/webpage.js" QMap(("mode", QVariant(QString, "r")))
URL: http://local.vagrant:1080/test/2/error.html?1459501847446
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 50
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 52
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 54
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 56
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 58
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 60
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 62
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 64
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 66
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 69
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 71
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 73
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 75
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 77
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 79
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 81
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 83
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 85
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 87
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 89
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
Preview: /vagrant/2/error_default.png
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/fs.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/system.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/webpage.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/fs.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/system.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] FileSystem - _open: ":/modules/webpage.js" QMap(("mode", QVariant(QString, "r")))
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 100
2016-04-01T12:10:47 [DEBUG] WebPage - updateLoadingProgress: 10
2016-04-01T12:10:47 [DEBUG] WebPage - setupFrame ""
2016-04-01T12:10:47 [DEBUG] Network - Resource request error: QNetworkReply::NetworkError(OperationCanceledError) ( "Operation canceled" ) URL: "http://local.vagrant:1080/test/2/css/style.css"
```

```
page.render('output.png');
```
generates nothing

But,
```
page.renderBase64('PNG');
```
returns content of transparent image with requested sizes.

I tested with PNG, JPG, GIF, PDF. None files of these formats were generated (with render function).

----------------------

Also, code hangs if specified image (in my case in <img> tag) not exists.

**Comment by Vitallium** [comment]
You didn't enable the `localToRemoteUrlAccessEnabled : "false"` option.

**Comment by VitalyOrbit** [comment]
I enabled this option in command line and from page.settings.
Pages renedered right if I call the command from console. If I call this command from php (exec, system and etc.) some files again not rendered.
I added onError for "page", but there weren't any errors.

I modify the page.content in a cycle and call render for every iteration. Just replace some texts.
Successfully rendered just first iteration file. Files for other iterations doesn't saved.

Can I do this way or page.content should not be changed?

And what the difference between launching from command line and launching from php script?

P.S. Phantom v 2.1.1 linux x64
